src/main/groovy/nextflow/executor/AwsBatchExecutor.groovy

```diff
@@ -318,16 +318,15 @@ class AwsBatchFileCopyStrategy extends SimpleFileCopyStrategy {
         """
         # aws helper
         nxf_s3_upload() {
-            local s3path="\${!#}"
-            local length=\$((\$#-1))
-            local files=\${@:1:\$length}
-            for f in \$files;do
+            local pattern=\$1
+            local s3path=\$2
+            for name in \$pattern;do
               if [[ -d \$f ]]; then
-                echo "aws s3 cp \$f \$s3path/\$f --quiet --storage-class $S3StorageClass --recursive $S3EncryptionCommand"
```

Comment:
It should be `if [[ -d \$name ]]; then`, right?

Reply:
Still running tests on Batch....I'll fix it.